src/test/java/org/opensearch/ad/rest/AnomalyDetectorRestApiIT.java

```diff
@@ -1504,4 +1508,393 @@ public void testValidateAnomalyDetectorWithWrongCategoryField() throws Exception
+            {
+                put("keyword-field", "keyword");
+                put("ip-field", "ip");
+            }
+        };
+        String testIndexData = "{\"keyword-field\": \"test-value\"}";
+        TestHelpers.createIndexWithHCADFields(client(), indexName, categoryFieldsAndTypes);
+        TestHelpers.ingestDataToIndex(client(), indexName, TestHelpers.toHttpEntity(testIndexData));
+        AnomalyDetector detector = createAnomalyDetector(
+            TestHelpers
+                .randomAnomalyDetectorUsingCategoryFields(
+                    randomAlphaOfLength(10),
+                    "timestamp",
+                    ImmutableList.of(indexName),
+                    categoryFieldsAndTypes.keySet().stream().collect(Collectors.toList())
+                ),
+            true,
+            client()
+        );
+
+        // Delete any existing result index
+        if (indexExists(CommonName.ANOMALY_RESULT_INDEX_ALIAS)) {
+            deleteIndex(CommonName.ANOMALY_RESULT_INDEX_ALIAS);
+        }
+        Response response = searchTopAnomalyResults(
+            detector.getDetectorId(),
+            false,
+            "{\"size\":3,\"category_field\":[\"keyword-field\"]," + "\"start_time_ms\":0, \"end_time_ms\":1}",
+            client()
+        );
+        Map<String, Object> responseMap = entityAsMap(response);
+        @SuppressWarnings("unchecked")
+        List<Map<String, Object>> buckets = (ArrayList<Map<String, Object>>) XContentMapValues.extractValue("buckets", responseMap);
+        assertEquals(0, buckets.size());
+    }
+
+    public void testSearchTopAnomalyResultsOnEmptyResultIndex() throws IOException {
+        String indexName = randomAlphaOfLength(10).toLowerCase();
+        Map<String, String> categoryFieldsAndTypes = new HashMap<String, String>() {
+            {
+                put("keyword-field", "keyword");
+                put("ip-field", "ip");
+            }
+        };
+        String testIndexData = "{\"keyword-field\": \"test-value\"}";
+        TestHelpers.createIndexWithHCADFields(client(), indexName, categoryFieldsAndTypes);
+        TestHelpers.ingestDataToIndex(client(), indexName, TestHelpers.toHttpEntity(testIndexData));
+        AnomalyDetector detector = createAnomalyDetector(
+            TestHelpers
+                .randomAnomalyDetectorUsingCategoryFields(
+                    randomAlphaOfLength(10),
+                    "timestamp",
+                    ImmutableList.of(indexName),
+                    categoryFieldsAndTypes.keySet().stream().collect(Collectors.toList())
+                ),
+            true,
+            client()
+        );
+
+        // Clear any existing result index, create an empty one
+        if (indexExists(CommonName.ANOMALY_RESULT_INDEX_ALIAS)) {
+            deleteIndex(CommonName.ANOMALY_RESULT_INDEX_ALIAS);
+        }
+        TestHelpers.createEmptyAnomalyResultIndex(client());
```

Comment:
Seems this is not the normal process from user's perspective. We don't allow user to create default AD result index directly. How about change to run historical analysis to create AD result index, then search top anomaly result?

Reply:
The purpose of this test was about having the index exist, but not return any results, so I just create the empty index so the API can query it, but still return empty results. Because this API doesn't interact with anything other than reading the results index, I still think this is ok. Same strategy is used in the below tests, as well as the existing search results tests.

I still prefer this than to run a historical analysis, so as to keep the test more lightweight, faster, and not increase the chances of it becoming flaky. Let me know if you still have concerns.

Reply:
Got it. Sounds good. 
Another way to generate an empty default result index is run historical analysis on a short period (so it can run quickly), then delete all results with delete result REST API. But like you said, these steps will make the test heavier and more flaky likely.